packages/spectral/src/serverTypes/asyncContext.ts

```diff
@@ -0,0 +1,23 @@
+import { AsyncLocalStorage } from "node:async_hooks";
+import { ActionContext } from "../types";
+
+const actionContextStorage = new AsyncLocalStorage<ActionContext>();
+
+export function getCurrentContext(): ActionContext | undefined {
+  return actionContextStorage.getStore();
+}
```

Comment:
Is there a reason we'd want to allow calling this without the `requireContext` checks or should we inline this in the `requireContext` function?